pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -1310,8 +1311,14 @@ def get_maximum_resources_available(self, chips=None, processor_id=None,
                 best_processor_id = self._best_core_available(
                     chip, key, processor_id)
                 processor = chip.get_processor_with_id(best_processor_id)
-                max_dtcm_available = processor.dtcm_available
-                max_cpu_available = processor.cpu_cycles_available
+                try:
+                    max_dtcm_available = processor.dtcm_available
+                    max_cpu_available = processor.cpu_cycles_available
+                except AttributeError as ae:
+                    if processor is None:
+                        raise PacmanProcessorNotAvailableError(
+                            chip_x, chip_y, best_processor_id)
+                    raise ae
```

Comment:
Note there is a python trick: if you just call raise without arguments, it re-raises the existing error, which means it keeps its stack trace.

Reply:
ok